app/assets/javascripts/discourse/helpers/category-link.js.es6

```diff
@@ -32,76 +38,12 @@ export function categoryBadgeHTML(category, opts) {
   )
     return "";
 
-  let description = get(category, "description_text");
-  let restricted = get(category, "read_restricted");
-  let url = opts.url
-    ? opts.url
-    : Discourse.getURL("/c/") + Discourse.Category.slugFor(category);
-  let href = opts.link === false ? "" : url;
-  let tagName = opts.link === false || opts.link === "false" ? "span" : "a";
-  let extraClasses = opts.extraClasses ? " " + opts.extraClasses : "";
-  let color = get(category, "color");
-  let html = "";
-  let parentCat = null;
-  let categoryDir = "";
-
-  if (!opts.hideParent) {
-    parentCat = Discourse.Category.findById(
-      get(category, "parent_category_id")
-    );
-  }
-
-  const categoryStyle =
-    opts.categoryStyle || Discourse.SiteSettings.category_style;
-  if (categoryStyle !== "none") {
-    if (parentCat && parentCat !== category) {
-      html += categoryStripe(
-        get(parentCat, "color"),
-        "badge-category-parent-bg"
-      );
+  for (let i = 0; i < _renderers.length; i++) {
```

Comment:
What's wrong with `_renderers.each(renderer => {`?

Reply:
You can probably be even fancier and do something like

```javascript
let result = _renderers.find(r => r.render(category, opts));
if (results) return result;
```

Reply:
I tried both options, but they didn't work well. But thanks for raising this, the loop there is unnecessary, all we need is to run the render of the first item in `_renderers`, so I switched to simply: 

```
return _renderers[0].render(category, opts);
```

Reply:
🤔 if we're only ever using only the first rendered, what's the point of allowing to register more than one?

Reply:
To allow theme components / plugins to override the default one. We use a similar pattern for icon renderers in `icon-library.js.es6`.

Reply:
But if it's an override, why keep them in an array?

Reply:
No specific reason. I'll take a look at refactoring.